api/v1alpha1/utils/commons.go

```diff
@@ -240,9 +283,39 @@ func (kc *KubeAuthConfiguration) createVaultClient(context context.Context, jwt
 
 	client.SetToken(secret.Auth.ClientToken)
 
+	go kc.startLifetimeWatcher(client, secret, log)
+
 	return client, nil
 }
 
+func (kc *KubeAuthConfiguration) startLifetimeWatcher(client *vault.Client, secret *vault.Secret, log logr.Logger) {
```

Comment:
can you explain why this is needed?

Reply:
If the TTL for the token is less than the maxTTL, the lifetime watcher renews the token until it reaches the maxTTL. Added a comment